miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -384,31 +391,40 @@ private void setChangeDetails(Box box) throws IOException {
   }
 
   @Override
-  public void updateBoxableLocation(Boxable boxable) throws IOException {
-    if (boxable.isDiscarded()) {
-      boxable.removeFromBox();
-    }
-    if (boxable.getBox() != null && boxable.getBoxPosition() == null) {
-      throw new IllegalArgumentException("Box position missing");
-    } else if (boxable.getBoxPosition() != null && (boxable.getBox() == null || boxable.getBox().getId() == AbstractBox.UNSAVED_ID)) {
-      throw new IllegalArgumentException("Box position set, but no box specified");
-    }
-    BoxableView managed = getBoxableView(new BoxableId(boxable.getEntityType(), boxable.getId()));
-    if (managed.getBoxId() != null && boxable.getBox() == null) {
-      Box box = getDetached(managed.getBoxId());
-      box.removeBoxable(managed.getBoxPosition());
-      save(box);
-    } else if (boxable.getBox() != null && (
-        managed.getBoxId() == null
-            || managed.getBoxId().longValue() != boxable.getBox().getId()
-            || !managed.getBoxPosition().equals(boxable.getBoxPosition())
-        )) {
-      Box box = getDetached(boxable.getBox().getId());
-      if (box.getBoxable(boxable.getBoxPosition()) != null) {
-        throw new IllegalArgumentException(String.format("Box position already occupied: %s %s", box.getName(), boxable.getBoxPosition()));
+  public void updateBoxableLocation(Boxable boxable, Boxable original) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)